deps/rabbit/src/rabbit_db_binding_m2k_converter.erl

```diff
@@ -29,90 +28,90 @@
+init_copy_to_khepri(_StoreId, _MigrationId, Tables) ->
     %% Clean up any previous attempt to copy the Mnesia table to Khepri.
     lists:foreach(fun clear_data_in_khepri/1, Tables),
 
-    State = #?MODULE{store_id = StoreId,
-                     exchanges = sets:new([{version, 2}])},
+    State = #?MODULE{},
     {ok, State}.
 
--spec copy_to_khepri(Table, Record, Priv) -> Ret when
+-spec copy_to_khepri(Table, Record, State) -> Ret when
       Table :: mnesia_to_khepri:mnesia_table(),
       Record :: tuple(),
-      Priv :: #?MODULE{},
-      Ret :: {ok, NewPriv} | {error, Reason},
-      NewPriv :: #?MODULE{},
+      State :: rabbit_db_m2k_converter:state(),
+      Ret :: {ok, NewState} | {error, Reason},
+      NewState :: rabbit_db_m2k_converter:state(),
       Reason :: any().
 %% @private
 
 copy_to_khepri(rabbit_route = Table,
                #route{binding = #binding{source = XName} = Binding},
-               #?MODULE{exchanges = Xs0} = State) ->
+               State) ->
     ?LOG_DEBUG(
        "Mnesia->Khepri data copy: [~0p] key: ~0p",
        [Table, Binding],
        #{domain => ?KMM_M2K_TABLE_COPY_LOG_DOMAIN}),
+    StoreId = rabbit_db_m2k_converter:get_store_id(State),
     Path = rabbit_db_binding:khepri_route_path(Binding),
-    ?LOG_DEBUG(
-       "Mnesia->Khepri data copy: [~0p] path: ~0p",
-       [Table, Path],
-       #{domain => ?KMM_M2K_TABLE_COPY_LOG_DOMAIN}),
-    rabbit_khepri:transaction(
-      fun() ->
-              Xs = case sets:is_element(XName, Xs0) of
-                       true ->
-                           Xs0;
-                       false ->
-                           %% If the binding's source is a new exchange,
-                           %% store the exchange's type in the exchange
-                           %% name branch of the tree.
-                           XPath = rabbit_db_binding:khepri_route_exchange_path(XName),
-                           [#exchange{type = XType}] =
-                               rabbit_db_exchange:get_in_khepri_tx(XName),
-                           ok = khepri_tx:put(XPath, #{type => XType}),
-                           sets:add_element(XName, Xs0)
-                   end,
-              Set = case khepri_tx:get(Path) of
-                        {ok, Set0} ->
-                            Set0;
-                        _ ->
-                            sets:new([{version, 2}])
-                    end,
-              case khepri_tx:put(Path, sets:add_element(Binding, Set)) of
-                  ok    -> {ok, State#?MODULE{exchanges = Xs}};
-                 Error -> Error
-              end
-      end, rw);
+    rabbit_db_m2k_converter:with_correlation_id(
+      fun(CorrId) ->
+              Extra = #{async => CorrId},
+              ?LOG_DEBUG(
+                 "Mnesia->Khepri data copy: [~0p] path: ~0p corr: ~0p",
+                 [Table, Path, CorrId],
+                 #{domain => ?KMM_M2K_TABLE_COPY_LOG_DOMAIN}),
+              khepri:transaction(
```

Comment:
I think I would prefer we always use the `rabbit_khepri` wrapper in case we want to do generic things in the future.

The comment above about the store identifier becomes irrelevant if we do this.